packages/vite/src/index.ts

```diff
@@ -0,0 +1,110 @@
+import { realpathSync, readFileSync } from 'fs';
+import { copyFile } from 'fs/promises';
+import { join, parse } from 'path';
+import { AppMeta, PackagerInstance, Variant } from '@embroider/core';
+import { build, BuildOptions, InlineConfig } from 'vite';
+import templateCompilerPlugin from '@embroider/rollup-plugin-hbs';
+import { babel } from '@rollup/plugin-babel';
+import commonjs from '@rollup/plugin-commonjs';
+
+type RollupOptions = Required<BuildOptions['rollupOptions']>;
+type AllowedRollupOptions = Omit<RollupOptions, 'input'>;
+
+type AllowedBuildOptions = Omit<BuildOptions, 'outDir' | 'rollupOptions'> & {
+  rollupOptions?: AllowedRollupOptions;
+};
+
+type AllowedViteConfig = Omit<InlineConfig, 'base' | 'build' | 'mode' | 'resolve' | 'root'> & {
+  build?: AllowedBuildOptions;
+};
```

Comment:
This might all be over-kill, but I wanted to make sure that the configuration options reflect which options are set internally and which can be configured externally.